### ISSUE TITLE
CSS modules and server rendering

### ISSUE THREAD

**Issue by razdvapoka** [redfin/react-server#689]
Hi!
I'm currently trying to use CSS modules in a react-server project and I've stumbled upon  the following issue: when I open the page in the browser I see this warning:

warning.js?8a56:36Warning: React attempted to reuse markup in a container but the checksum was invalid. This generally means that you are using server rendering and the markup generated on the server was not what the client was expecting. React injected new markup to compensate which works but you have lost many of the benefits of server rendering. Instead, figure out why the markup being generated is different on the client or server:
 (client) yle="margin-top:0;" class="App--containe
 (server) yle="margin-top:0;" data-reactroot="" da

This is clearly due to the server being not able to get css classes' names while rendering the components.

Any ideas how to approach this issue? My colleague pointed out using https://github.com/kriasoft/isomorphic-style-loader, but I have no idea how to provide context for components using react-server's page API. 

I saw @hzhu's attempt to integrate css modules here: https://github.com/hzhu/react-server-redux/commit/5cb4831b68d173582b0da3e937020db5183a2d4b, cloned and launched his project, but it has the same issue.

**Comment by karlhorky** [comment]
I guess this will be enabled by #791?

**Comment by jhnns** [comment]
That's correct, you will also need to compile the server code with webpack to make this work.

My usual configuration for this looks like:

```javascript
            {
                test: /\.css$/,
                loader: ExtractTextPlugin.extract(
                    'style-loader',
                    `css-loader${ isNodeTarget ? '/locals' : '' }?` + JSON.stringify({
                        modules: true
                     })
                )
            },
```

```javascript
        new ExtractTextPlugin('[name].[contenthash].css', {
            disable: isNodeTarget || isDev
        }),
```
